c10/core/DispatchKeySet.cpp

```diff
@@ -46,8 +46,14 @@ bool isBackendDispatchKey(DispatchKey t) {
 // math_dispatch_keyset contains all keys in backend_dispatch_keyset and
 // autograd_dispatch_keyset Alias key DispatchKey::CompositeImplicitAutograd
 // maps to [math_dispatch_keyset x full_backend_mask]
-constexpr DispatchKeySet math_dispatch_keyset =
-    backend_dispatch_keyset | autograd_dispatch_keyset;
+constexpr DispatchKeySet math_dispatch_keyset = backend_dispatch_keyset |
+    autograd_dispatch_keyset |
+    // See Note [NestedTensor Not Included in Backend Keys]
+    // The caveat to that note is that nested_tensor is a special case
+    // where we would like to support composite implict kernels but not
+    // explicit kernels therefore we manually add the key to the
```

Comment:
I'm not sure to understand? Why can't you just override the explicit kernels you don't want?

Reply:
This is Driss's work that we thought would be a necessary fix here. The original discussion is [here](https://github.com/pytorch/pytorch/issues/79048)

@drisspg @bdhirsh would know better than I do

Reply:
This was added in order to allow for composite implicit kernels to work with nested_tensors. Some functions like softmax are composite implicit functions that in turn call non composite ops like `_softmax `. In order to be able to register at _softmax the nested tensor needs to be able to run by composite function.  Similiarly Reshape as an example of one of these ops whose top most function is composite implicit that quickly calls a non-composite op

Reply:
I understand that you want it to get the Implicit kernels, but this comment seem to indicate that you do special logic (compared to the other classical backend keys) to prevent it from getting the Explicit kernels. Did I read this comment wrong?

Reply:
Yeah that is correct, work for implicit don't work for explicit. I went back and fourth about this, practically it appears that there are alot more composite explicit keys that end up calling sizes and  producing strange hard to debug errors for users. This possibility exists for composite implicit as well, however was happening with less frequency for those kernels.

Is your suggestion to enable for both composite kernels types and whenever we find a composite explicit doing something wrong just register a Torch_check(false) kernel?

Reply:
> Is your suggestion to enable for both composite kernels types and whenever we find a composite explicit doing something wrong just register a Torch_check(false) kernel?

That would have been my first guess yes.
But I am not sure:
- how involved is the current "workaround" to get Implicit but not Explicit kernels? cc @bdhirsh 
- What happens when you want to use some of the Explicit kernels? You now have to copy/paste them?